### PR TITLE
🐛 Fixed 404 routing in API

### LINKED ISSUE #10020: GET /ghost/api/v3/admin redirects to /ghost#
This is due to how the routers are mounted, if we don't exactly match the path of an api, we redirect to the admin, this is unexpected behaviour!

We should create router for the `/api` mount point and 404 any unmatched paths!

### PR DESCRIPTION
**Pull request by anaplian**
closes #10020

I added a default API application that always returns a 404 error response. This is so that requests that don't get routed to any particular API are given a properly JSON-formatted 404 error (previously they were routed to the admin app's 404 page.)

- [x] There's a clear use-case for this code change
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `npm test`)

### CODE REVIEW

**Comment by kirrg001** [comment]
Thanks will review on Monday :)

**Review by kirrg001 — CHANGES_REQUESTED** [review]
Thanks for your PR. I've left a comment :)

**Comment by kirrg001** [comment]
@anaplian ping :)

**Comment by anaplian** [comment]
Thanks for the feedback :slightly_smiling_face: 
Working on it now!

File: core/server/web/api/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const errorHandler = require('../shared/middlewares/error-handler');
+
+module.exports = function setupApiApp() {
+    debug('Parent API setup start');
+    const apiApp = express();
+
+    // Mount different API versions
+    apiApp.use(urlUtils.getVersionPath({version: 'v0.1'}), require('./v0.1/app')());
+    apiApp.use(urlUtils.getVersionPath({version: 'v2', type: 'content'}), require('./v2/content/app')());
+    apiApp.use(urlUtils.getVersionPath({version: 'v2', type: 'admin'}), require('./v2/admin/app')());
```
Comment (kirrg001) [inline core/server/web/api/index.js]: The PR looks great 👍 
Was wondering why we mount urls without trailing slash now? 🤔

Comment (kirrg001) [reply core/server/web/api/index.js]: Was resolved in an extra commit 👍

**Comment by kirrg001** [comment]
@allouis Could you pls double check/test this PR on Monday? I think this one is good to go 👍

**Review by kirrg001 — APPROVED** [review]

**Comment by gargol** [comment]
@allouis lmk if we can merge this PR in for tomorrow's release :wink:

**Pull request merged**